### PR TITLE
Moving ColourMsg to Common and adding to miner

### PR DESCRIPTION
**Pull request by arbourp**
Adding to the miner so it always generates a text when it finds a block.
Also, making it colourful.

Add small note about building in Windows with CMAKE and Boost Versions.

### CODE REVIEW

**Comment by michael-herwig** [comment]
You have some merge conflicts could you please [rebase](https://git-scm.com/docs/git-rebase) the `dev` branch and push again?

File: src/Miner/Miner.cpp [hunk 1]
```diff
@@ -98,6 +100,7 @@ void Miner::workerFunc(const BlockTemplate& blockTemplate, uint64_t difficulty,
       CachedBlock cachedBlock(block);
       Crypto::Hash hash = cachedBlock.getBlockLongHash();
       if (check_hash(hash, difficulty)) {
+	std::cout << InformationMsg("Found block for difficulty ")  << difficulty  << std::endl;
```
Comment (michael-herwig) [inline src/Miner/Miner.cpp:103]: We should not write directly to std::cout, whe have the `Logging::ILogger` interface for that. May consider extending the interface if it does not fit your need.

**Review by michael-herwig — CHANGES_REQUESTED** [review]
The Build.md has moved to our wiki, maybe @Neo-Amity or @19morpheus80 can merge your changes? My english knowledge is not good enough:/

File: src/Miner/Miner.cpp (hunk 1 above)
Comment (arbourp) [reply src/Miner/Miner.cpp:103]: Okay, good point, if we are wanting to use the ILogger I'll use that class for the future.

**Review by 19morpheus80 — APPROVED** [review]

**Pull request closed without merging**